mlflow/prompt/registry_utils.py

```diff
@@ -96,3 +102,45 @@ def wrapper(*args, **kwargs):
                 raise e
 
     return wrapper
+
+
+def validate_prompt_name(name: Any):
+    """Validate the prompt name against the prompt specific rule"""
+    if not isinstance(name, str) or not name:
+        raise MlflowException.invalid_parameter_value(
+            "Prompt name must be a non-empty string.",
+        )
+
+    if PROMPT_NAME_RULE.match(name) is None:
+        raise MlflowException.invalid_parameter_value(
+            "Prompt name can only contain alphanumeric characters, hyphens, underscores, and dots.",
+        )
+
+
+def handle_resource_already_exist_error(
+    name: str,
+    is_existing_entity_prompt: bool,
+    is_new_entity_prompt: bool,
+):
+    """
+    Show a more specific error message for name conflict in Model Registry.
+
+    1. When creating a model with the same name as an existing model, say "model already exists".
+    2. When creating a prompt with the same name as an existing prompt, say "prompt already exists".
+    3. Otherwise, explain that a prompt and a model cannot have the same name.
+    """
+    old_entity = "Prompt" if is_existing_entity_prompt else "Registered Model"
+    new_entity = "Prompt" if is_new_entity_prompt else "Registered Model"
+
+    if old_entity != new_entity:
```

Comment:
nit, can probably be simplified
```suggestion
    if is_existing_entity_prompt != is_new_entity_prompt:
```

Reply:
We use the `old_entity` and `new_entity` in the error message below:)